docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,209 @@
+    - List of NS to filtered based on a LabelSelector `(Spec.features.nonAdmin.labelSelector)`
+    - By default, of none of the above are specified then NAC listens to all the Namespaces in the cluster
+    - We need to error out if NAC NS scopes overlap (i.e. NS watch list overlap)
+
+
+## Pre-requisites
+- **OADP installed**: OADP Operator must be installed and configured to use non-admin controller
+- **Non Admin Controller configured**: Data Protection Application (DPA) instance must configure Non Admin Controller to watch user namespace(s), by default it watches all Namespaces
+- **RBAC privileges for the user**: User must have the appropriate RBAC privileges to create Non Admin objects within the Namespace where Backup will be taken. An example of such ClusterRole, which may be added to the user with `RoleBinding`:
+    ```yaml
+    # permissions for end users to edit non-admin custom resources.
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+    labels:
+        app.kubernetes.io/name: clusterrole
+        app.kubernetes.io/instance: nonadminbackup-editor-role
+        app.kubernetes.io/component: rbac
+        app.kubernetes.io/created-by: oadp-nac
+        app.kubernetes.io/part-of: oadp-nac
+        app.kubernetes.io/managed-by: kustomize
+    name: nonadminbackup-editor-role
+    rules:
+    - apiGroups:
+        - nac.oadp.openshift.io
+      resources:
+        - nonadminbackups
+        - nonadminrestores
+        - nonadminbackupstorageloations
+      verbs:
+        - create
+        - delete
+        - get
+        - list
+        - patch
+        - update
+        - watch
+    - apiGroups:
+        - nac.oadp.openshift.io
+      resources:
+        - nonadminbackups/status
+        - nonadminrestores/status
+        - nonadminbackupstoragelocations/status
+      verbs:
+        - get
+
+    ```
+- **Existence of Namespace:** The existence of namespace prior to performing non-admin backup/restore operations is a must.(for backup it will obviously exist but for restore it is an explicit hard precondition)
+## High-Level design
+
+### Components
+- **OADP Operator:** OADP is the OpenShift API for Data Protection operator. This open source operator sets up and installs Velero on the OpenShift platform, allowing users to backup and restore applications.
+- **Controllers:** The Non-Admin controller will pack the following controllers as part of it:
+    - **Non-Admin Backup (NAB) Controller:** The responsibilities of the NAB controller are:
+        - Listen to requests pertaining to Non-Admin Backup CRD
+        - Process requests pertaining to Non-Admin Backup CRD
+        - Update Non-Admin Backup CR status with the status/events from Velero Backup CR
+        - Cascade Any actions performed on Non-Admin Backup CR to corresponding Velero backup CR
+    - **Non-Admin Restore (NAR) Controller:** The responsibilities of the NAR controller are:
```

Comment:
Create Velero Restore...

Reply:
yeah `cascade any actions` should cover that